src/Microsoft.ML.Core/Data/ColumnType.cs

```diff
@@ -554,43 +554,47 @@ public override bool Equals(ColumnType other)
     /// class numbers, etc. For example, in multi-class classification, the label is typically
     /// a class number which is naturally a KeyType.
     ///
-    /// KeyTypes can be contiguous (the class number example), in which case they can have
-    /// a cardinality/Count. For non-contiguous KeyTypes the Count property returns zero.
-    /// Any KeyType (contiguous or not) can have a Min value. The Min value is always >= 0.
+    /// KeyTypes with a known range of values have a cardinality/Count. If the range is
+    /// unknown the Count property returns zero.
     ///
     /// Note that the representation value does not necessarily match the logical value.
-    /// For example, if a KeyType has range 1000-5000, then it has a Min of 1000, Count
-    /// of 4001, but the representational values are 1-4001. The representation value zero
-    /// is reserved to mean none/invalid.
+    /// For example, if a KeyType has range 0-5000, then it has a Count of 5001, but
+    /// the representational values are 1-5001. The representation value zero is reserved
+    /// to mean none/invalid.
     /// </summary>
     public sealed class KeyType : PrimitiveType
     {
-        private KeyType(Type type, DataKind kind, ulong min, int count, bool contiguous)
+        private KeyType(Type type, DataKind kind, ulong count)
             : base(type, kind)
         {
             Contracts.AssertValue(type);
             Contracts.Assert(kind.ToType() == type);
+            Count = count;
+        }
 
-            Contracts.CheckParam(min >= 0, nameof(min));
-            Contracts.CheckParam(count >= 0, nameof(count), "Must be non-negative.");
-            Contracts.CheckParam((ulong)count <= ulong.MaxValue - min, nameof(count));
-            Contracts.CheckParam((ulong)count <= kind.ToMaxInt(), nameof(count));
-            Contracts.CheckParam(contiguous || count == 0, nameof(count), "Must be 0 for non-contiguous");
+        public KeyType(Type type, ulong count)
+            : this(type, CheckRefRawType(type), count)
+        {
+        }
 
-            Contiguous = contiguous;
-            Min = min;
-            Count = count;
+        [BestFriend]
+        internal KeyType(Type type, int count)
+            : this(type, CheckRefRawType(type), (ulong)count)
+        {
+            Contracts.Assert(0 <= count);
         }
 
-        public KeyType(Type type, ulong min, int count, bool contiguous = true)
-            : this(type, CheckRefRawType(type), min, count, contiguous)
+        [BestFriend]
+        internal KeyType(DataKind kind, ulong count)
+            : this(ToRawType(kind), kind, count)
         {
         }
 
         [BestFriend]
-        internal KeyType(DataKind kind, ulong min, int count, bool contiguous = true)
-            : this(ToRawType(kind), kind, min, count, contiguous)
+        internal KeyType(DataKind kind, int count)
+            : this(ToRawType(kind), kind, (ulong)count)
         {
+            Contracts.Assert(0 <= count);
```

Comment:
>Assert [](start = 22, length = 6)

Should this be a check?

Reply:
Yes, we probably should not even have it at all though.

---
In reply to: [248102911](https://github.com/dotnet/machinelearning/pull/2146#discussion_r248102911) [](ancestors = 248102911)

Reply:
Thanks for pointing out the difference between Assert and Check, I was ignoring it and makes much more sense like this. 

---
In reply to: [248116058](https://github.com/dotnet/machinelearning/pull/2146#discussion_r248116058) [](ancestors = 248116058,248102911)